### PR TITLE
update the "manifest lists" test for k8s.gcr.io

### PR DESCRIPTION
**Pull request by neolit123**
commits:

> tests/e2e/manifests: use k8s.gcr.io as the GCR bucket

only switches the URL for the bucket

> tests/e2e/manifests: pin the go version and install it using curl

installing go from `debian:stretch` suddenly started returning a version that does not have `go mod`.
download go and install gcc manually.

/assign @SataQiu 
fixes: https://github.com/kubernetes/kubeadm/issues/2051

### CODE REVIEW

File: tests/e2e/manifests/verify_manifest_lists.sh [hunk 1]
```diff
@@ -16,21 +16,29 @@ trap "rm go.mod go.sum" EXIT
 
 # install curl if missing
 if ! `curl --version > /dev/null`; then
-	apt-get update || exit 1
-	apt-get install -y curl || exit 1
```
Comment (neolit123) [inline tests/e2e/manifests/verify_manifest_lists.sh]: `|| exit 1` is not needed due to `set -o errexit`

File: tests/e2e/manifests/verify_manifest_lists.sh [hunk 2]
```diff
@@ -16,21 +16,29 @@ trap "rm go.mod go.sum" EXIT
...
+#   exec: "gcc": executable file not found in $PATH
+if ! `gcc --v > /dev/null`; then
+	apt-get install -y gcc
 fi
 
 LPATH=`dirname "$0"`
 cd "$LPATH"
 
 # use go modules. this forces using the latest k8s.io/apimachinery package.
-export GO111MODULE=on
```
Comment (neolit123) [inline tests/e2e/manifests/verify_manifest_lists.sh]: not needed for go 1.13*

File: tests/e2e/manifests/verify_manifest_lists.sh [hunk 3]
```diff
@@ -16,21 +16,29 @@ trap "rm go.mod go.sum" EXIT
...
+	curl https://dl.google.com/go/go1.13.8.linux-amd64.tar.gz -o /tmp/go.tar.gz
+	tar -C /usr/local -xzf /tmp/go.tar.gz
+	export PATH="$PATH":/usr/local/go/bin
+fi
+
+# api-machinery requires gcc
+#   go: extracting k8s.io/apimachinery v0.17.3
+#   runtime/cgo
+#   exec: "gcc": executable file not found in $PATH
+if ! `gcc --v > /dev/null`; then
```
Comment (SataQiu) [inline tests/e2e/manifests/verify_manifest_lists.sh]: LGTM @neolit123 ! 
Just a nit: `gcc --v` doesn't seem like a correct command:
```sh
# gcc --v
gcc: error: unrecognized command line option ‘--v’
gcc: fatal error: no input files
# gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/4.8.5/lto-wrapper
Target: x86_64-redhat-linux
Configured with: ../configure --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-bootstrap --enable-shared --enable-threads=posix --enable-checking=release --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-linker-hash-style=gnu --enable-languages=c,c++,objc,obj-c++,java,fortran,ada,go,lto --enable-plugin --enable-initfini-array --disable-libgcj --with-isl=/builddir/build/BUILD/gcc-4.8.5-20150702/obj-x86_64-redhat-linux/isl-install --with-cloog=/builddir/build/BUILD/gcc-4.8.5-20150702/obj-x86_64-redhat-linux/cloog-install --enable-gnu-indirect-function --with-tune=generic --with-arch_32=x86-64 --build=x86_64-redhat-linux
Thread model: posix
gcc version 4.8.5 20150623 (Red Hat 4.8.5-36) (GCC) 
```

Comment (neolit123) [reply tests/e2e/manifests/verify_manifest_lists.sh]: correct, it's a typo. will update in a bit.

Comment (neolit123) [reply tests/e2e/manifests/verify_manifest_lists.sh]: updated.

**Pull request merged**